spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/MongoDbContainerTest.java

```diff
@@ -40,53 +42,68 @@
 import org.springframework.integration.mongodb.outbound.MessageCollectionCallback;
 import org.springframework.messaging.Message;
 
+import com.mongodb.ConnectionString;
 import com.mongodb.MongoClientSettings;
 import com.mongodb.MongoException;
 import com.mongodb.client.MongoClients;
 import com.mongodb.client.MongoCollection;
 
-
 /**
- * Convenience base class that enables unit test methods to rely upon the {@link MongoDbAvailable} annotation.
+ * The base contract for all tests requiring a MongoDb connection.
+ * The Testcontainers 'reuse' option must be disabled, so, Ryuk container is started
+ * and will clean all the containers up from this test suite after JVM exit.
+ * Since the Redis container instance is shared via static property, it is going to be
+ * started only once per JVM, therefore the target Docker container is reused automatically.
  *
  * @author Oleg Zhurakousky
  * @author Xavier Padro
  * @author Artem Bilan
  * @author David Turanski
+ * @author Artem Vozhdayenko
  *
- * @since 2.1
+ * @since 6.0
  */
-public abstract class MongoDbAvailableTests {
-
-	@Rule
-	public MongoDbAvailableRule mongoDbAvailableRule = new MongoDbAvailableRule();
-
-	public static final MongoDatabaseFactory MONGO_DATABASE_FACTORY =
-			new SimpleMongoClientDatabaseFactory(
-					MongoClients.create(
-							MongoClientSettings.builder().uuidRepresentation(UuidRepresentation.STANDARD).build()),
-					"test");
-
-	public static final ReactiveMongoDatabaseFactory REACTIVE_MONGO_DATABASE_FACTORY =
-			new SimpleReactiveMongoDatabaseFactory(
-					com.mongodb.reactivestreams.client.MongoClients.create(
-							MongoClientSettings.builder().uuidRepresentation(UuidRepresentation.STANDARD).build()),
-					"test");
-
-	protected MongoDatabaseFactory prepareMongoFactory(String... additionalCollectionsToDrop) {
-		cleanupCollections(MONGO_DATABASE_FACTORY, additionalCollectionsToDrop);
-		return MONGO_DATABASE_FACTORY;
+@Testcontainers(disabledWithoutDocker = true)
+public interface MongoDbContainerTest {
+	GenericContainer<?> MONGO_CONTAINER = new GenericContainer<>("mongo:5.0.9")
```

Comment:
It's better to have every class member surrounded with blank line: increases source code readability.